core/cppgc.rs

```diff
@@ -41,8 +41,9 @@ pub fn make_cppgc_object<'a, T: 'static>(
   obj
 }
 
+#[allow(clippy::needless_lifetimes)]
```

Comment:
Is this ignore actually needed here?

Reply:
No, I could remove the lifetimes. But they are very important to the correctness of the function, so it's better to keep them explicit